diff --git a/toolkit/moz.configure b/toolkit/moz.configure
index e5e517635d63..cc7aaf16ffcc 100644
--- a/toolkit/moz.configure
+++ b/toolkit/moz.configure
@@ -3662,8 +3662,8 @@ with only_when(compile_environment):
     @depends(target)
     def default_user_appdir(target):
         if target.kernel in ("WINNT", "Darwin"):
-            return "Mozilla"
-        return ".mozilla"
+            return "Librewolf"
+        return ".librewolf"
 
     option(
         "--with-user-appdir",
