diff --git a/browser/components/ProfileDataUpgrader.sys.mjs b/browser/components/ProfileDataUpgrader.sys.mjs
index 6cee3dc..dd91621 100644
--- a/browser/components/ProfileDataUpgrader.sys.mjs
+++ b/browser/components/ProfileDataUpgrader.sys.mjs
@@ -1076,6 +1076,12 @@ export let ProfileDataUpgrader = {
 
       Services.prefs.clearUserBranch("cookiebanners.");
       Services.prefs.clearUserBranch("browser.promo.cookiebanners.");
+
+      if (Services.prefs.prefHasUserValue("media.eme.enabled")) {
+        Services.prefs.setBoolPref("media.gmp-manager.updateEnabled", true);
+        Services.prefs.setBoolPref("media.gmp-widevinecdm.enabled", true);
+        Services.prefs.setBoolPref("media.gmp-widevinecdm.autoupdate", true);
+      }
     }
 
     if (existingDataVersion < 180) {
