diff --git a/dom/canvas/CanvasRenderingContext2D.cpp b/dom/canvas/CanvasRenderingContext2D.cpp
index 7aaa3a6f82a9..dbd67196e375 100644
--- a/dom/canvas/CanvasRenderingContext2D.cpp
+++ b/dom/canvas/CanvasRenderingContext2D.cpp
@@ -2285,7 +2285,9 @@ UniquePtr<uint8_t[]> CanvasRenderingContext2D::GetImageBuffer(
         PrincipalOrNull(), ret.get(), out_imageSize->width,
         out_imageSize->height,
         out_imageSize->width * out_imageSize->height * 4);
-    if (aExtractionBehavior == CanvasUtils::ImageExtraction::Randomize) {
+    if (aExtractionBehavior == CanvasUtils::ImageExtraction::Randomize ||
+        aExtractionBehavior ==
+            CanvasUtils::ImageExtraction::EfficientRandomize) {
       nsRFPService::RandomizePixels(
           GetCookieJarSettings(), PrincipalOrNull(), ret.get(),
           out_imageSize->width, out_imageSize->height,
@@ -6612,7 +6614,8 @@ nsresult CanvasRenderingContext2D::GetImageDataArray(
   //
   // Note that we don't need to clone if we will use the place holder because
   // the place holder doesn't use actual image data.
-  if (extractionBehavior == CanvasUtils::ImageExtraction::Randomize) {
+  if (extractionBehavior == CanvasUtils::ImageExtraction::Randomize ||
+      extractionBehavior == CanvasUtils::ImageExtraction::EfficientRandomize) {
     if (readback) {
       readback = CreateDataSourceSurfaceByCloning(readback);
     }
@@ -6634,7 +6637,9 @@ nsresult CanvasRenderingContext2D::GetImageDataArray(
       // service) after we call JS_GetUint8ClampedArrayData, we will
       // pre-generate the randomness required for GeneratePlaceholderCanvasData.
       randomData = TryToGenerateRandomDataForPlaceholderCanvasData();
-    } else if (extractionBehavior == CanvasUtils::ImageExtraction::Randomize) {
+    } else if (extractionBehavior == CanvasUtils::ImageExtraction::Randomize ||
+               extractionBehavior ==
+                   CanvasUtils::ImageExtraction::EfficientRandomize) {
       // Apply the random noises if canvan randomization is enabled. We don't
       // need to calculate random noises if we are going to use the place
       // holder.
diff --git a/dom/canvas/ClientWebGLContext.cpp b/dom/canvas/ClientWebGLContext.cpp
index a9e56030607f..06253c113a0c 100644
--- a/dom/canvas/ClientWebGLContext.cpp
+++ b/dom/canvas/ClientWebGLContext.cpp
@@ -1362,7 +1362,8 @@ UniquePtr<uint8_t[]> ClientWebGLContext::GetImageBuffer(
   *out_imageSize = dataSurface->GetSize();
 
   nsRFPService::PotentiallyDumpImage(PrincipalOrNull(), dataSurface);
-  if (aExtractionBehavior == CanvasUtils::ImageExtraction::Randomize) {
+  if (aExtractionBehavior == CanvasUtils::ImageExtraction::Randomize ||
+      aExtractionBehavior == CanvasUtils::ImageExtraction::EfficientRandomize) {
     return gfxUtils::GetImageBufferWithRandomNoise(
         dataSurface, premultAlpha, GetCookieJarSettings(), PrincipalOrNull(),
         out_format);
@@ -3632,7 +3633,9 @@ void ClientWebGLContext::GetBufferSubData(GLenum target, GLintptr srcByteOffset,
       if (extraction == CanvasUtils::ImageExtraction::Placeholder) {
         dom::GeneratePlaceholderCanvasData(destView->size_bytes(),
                                            destView->Elements());
-      } else if (extraction == CanvasUtils::ImageExtraction::Randomize) {
+      } else if (extraction == CanvasUtils::ImageExtraction::Randomize ||
+                 extraction ==
+                     CanvasUtils::ImageExtraction::EfficientRandomize) {
         // We have no idea what's in the buffer. So, we randomize it as if each
         // elemSize bytes is a single element.
         uint8_t elementsPerGroup = 1,
@@ -5391,7 +5394,8 @@ void ClientWebGLContext::ReadPixels(GLint x, GLint y, GLsizei width,
       } else {
         RecordCanvasUsage(CanvasExtractionAPI::ReadPixels,
                           CSSIntSize(width, height));
-        if (extraction == CanvasUtils::ImageExtraction::Randomize) {
+        if (extraction == CanvasUtils::ImageExtraction::Randomize ||
+            extraction == CanvasUtils::ImageExtraction::EfficientRandomize) {
           const auto pii = webgl::PackingInfoInfo::For(desc.pi);
           // DoReadPixels() requres pii to be Some().
           MOZ_ASSERT(pii.isSome());
diff --git a/dom/canvas/OffscreenCanvasDisplayHelper.cpp b/dom/canvas/OffscreenCanvasDisplayHelper.cpp
index fddca1cd46f6..4dafeeaf0c7e 100644
--- a/dom/canvas/OffscreenCanvasDisplayHelper.cpp
+++ b/dom/canvas/OffscreenCanvasDisplayHelper.cpp
@@ -615,7 +615,8 @@ UniquePtr<uint8_t[]> OffscreenCanvasDisplayHelper::GetImageBuffer(
       principal, imageBuffer.get(), dataSurface->GetSize().width,
       dataSurface->GetSize().height,
       dataSurface->GetSize().width * dataSurface->GetSize().height * 4);
-  if (aExtractionBehavior == CanvasUtils::ImageExtraction::Randomize) {
+  if (aExtractionBehavior == CanvasUtils::ImageExtraction::Randomize ||
+      aExtractionBehavior == CanvasUtils::ImageExtraction::EfficientRandomize) {
     nsRFPService::RandomizePixels(
         cookieJarSettings, principal, imageBuffer.get(),
         dataSurface->GetSize().width, dataSurface->GetSize().height,
diff --git a/toolkit/components/resistfingerprinting/nsRFPService.cpp b/toolkit/components/resistfingerprinting/nsRFPService.cpp
index 9554fbb..0c1e380 100644
--- a/toolkit/components/resistfingerprinting/nsRFPService.cpp
+++ b/toolkit/components/resistfingerprinting/nsRFPService.cpp
@@ -1805,6 +1805,7 @@ nsresult nsRFPService::RandomizeElements(
     return NS_OK;
   }
 
+  /*
   // Don't randomize if all groups are uniform.
   const bool allGroupsMatch = [&]() {
     auto itr = RangedPtr<const uint8_t>(aData, aSizeInBytes);
@@ -1819,6 +1820,7 @@ nsresult nsRFPService::RandomizeElements(
   if (allGroupsMatch) {
     return NS_OK;
   }
+  */
 
   auto timerId =
       glean::fingerprinting_protection::canvas_noise_calculate_time_2.Start();
