diff --git a/browser/base/content/main-popupset.inc.xhtml b/browser/base/content/main-popupset.inc.xhtml
index 78030413c4..301e5afabf 100644
--- a/browser/base/content/main-popupset.inc.xhtml
+++ b/browser/base/content/main-popupset.inc.xhtml
@@ -782,6 +782,8 @@
               checked="false"
               autocheck="false"
               id="translations-panel-settings-never-translate-site"/>
+    <menuitem data-l10n-id="translations-panel-disable-translations"
+              id="translations-panel-disable-translations"/>
     <menuseparator/>
     <menuitem class="manage-languages-menuitem"
               data-l10n-id="translations-panel-settings-manage-languages"
diff --git a/browser/base/content/main-popupset.js b/browser/base/content/main-popupset.js
index 244ff30700..57a0d5a598 100644
--- a/browser/base/content/main-popupset.js
+++ b/browser/base/content/main-popupset.js
@@ -346,6 +346,9 @@ document.addEventListener(
         case "translations-panel-settings-never-translate-site":
           FullPageTranslationsPanel.onNeverTranslateSite();
           break;
+        case "translations-panel-disable-translations":
+          FullPageTranslationsPanel.openDisableTranslations();
+          break;
         case "translations-panel-manage-languages":
           FullPageTranslationsPanel.openManageLanguages();
           break;
diff --git a/browser/components/preferences/main.inc.xhtml b/browser/components/preferences/main.inc.xhtml
index ff577c035f..3b07de78a6 100644
--- a/browser/components/preferences/main.inc.xhtml
+++ b/browser/components/preferences/main.inc.xhtml
@@ -28,6 +28,7 @@
 
   <vbox id="startupPageBox">
     <html:setting-group groupid="startup"/>
+    <html:a is="moz-support-link" support-page="session-restore" data-l10n-id="session-restore-learn-more" />
   </vbox>
 
 </groupbox>
@@ -366,8 +367,16 @@
           preference="layout.spellcheckDefault"/>
 
   <!-- Translations -->
-  <vbox id="translationsGroup" hidden="true" data-subcategory="translations">
+  <vbox id="translationsGroup" data-subcategory="translations">
     <label><html:h2 data-l10n-id="translations-manage-header"/></label>
+    <checkbox id="translations-manage-enable" 
+          data-l10n-id="translations-manage-enable" 
+          preference="browser.translations.enable"
+          data-subcategory="translations-enable" />
+    <vbox id="innerTranslationsGroup" hidden="true">
+    <checkbox id="translations-manage-autopopup" 
+              data-l10n-id="translations-manage-autopopup" 
+              preference="browser.translations.automaticallyPopup" />
     <hbox id="translations-manage-description" align="center">
       <description flex="1" data-l10n-id="translations-manage-intro-2"/>
       <button id="translations-manage-settings-button"
@@ -389,6 +398,7 @@
       <description id="translations-manage-error" hidden="true"></description>
     </vbox>
   </vbox>
+  </vbox>
 </groupbox>
 
 <!-- Files and Applications -->
diff --git a/browser/components/preferences/main.js b/browser/components/preferences/main.js
index 46283d3..8bd2352 100644
--- a/browser/components/preferences/main.js
+++ b/browser/components/preferences/main.js
@@ -170,6 +170,10 @@ Preferences.addAll([
   { id: "dom.ipc.processCount.web", type: "int" },
   { id: "layers.acceleration.disabled", type: "bool", inverted: true },
 
+  /* Fullpage Translations */
+  { id: "browser.translations.enable", type: "bool" },
+  { id: "browser.translations.automaticallyPopup", type: "bool" },
+
   // Files and Applications
   { id: "pref.downloads.disable_button.edit_actions", type: "bool" },
 
@@ -2037,6 +2041,10 @@ var gMainPane = {
       document.getElementById("checkSpelling"),
       () => this.writeCheckSpelling()
     );
+    Preferences.addSyncFromPrefListener(
+      document.getElementById("translations-manage-enable"),
+      () => this.readEnableTranslations()
+    );
     Preferences.addSyncFromPrefListener(
       document.getElementById("linkTargeting"),
       () => this.readLinkTarget()
@@ -2270,24 +2278,33 @@ var gMainPane = {
       Preferences.get("browser.display.document_color_use").value != 2;
   },
 
+  readEnableTranslations(skipInit = false) {
+    const translationsEnabled = Preferences.get("browser.translations.enable").value;
+    document.getElementById("innerTranslationsGroup").hidden = !translationsEnabled;
+    if (!this._translationsInitialized && !skipInit)
+      this.initTranslations();
+  },
+
+  _translationsInitialized: false,
+
   /**
    * Initialize the translations view.
    */
   async initTranslations() {
+    this.readEnableTranslations(true);
+    
     if (!Services.prefs.getBoolPref("browser.translations.enable")) {
       return;
     }
 
+    this._translationsInitialized = true;
+
     /**
      * Which phase a language download is in.
      *
      * @typedef {"downloaded" | "loading" | "uninstalled"} DownloadPhase
      */
 
-    // Immediately show the group so that the async load of the component does
-    // not cause the layout to jump. The group will be empty initially.
-    document.getElementById("translationsGroup").hidden = false;
-
     class TranslationsState {
       /**
        * The fully initialized state.
@@ -4530,6 +4547,22 @@ var gMainPane = {
     // they'll only visit URLs derived from that template (i.e. with %s
     // in the template replaced by the URL of the content being handled).
 
+    let inPrompt = false;
+    Preferences.get("browser.translations.enable").on("change", () => {
+      if(!Preferences.get("browser.translations.enable").value)
+      if(!inPrompt) {
+        inPrompt = true;
+        confirmRestartPrompt(false, 1, true, false).then(buttonIndex => {
+          inPrompt = false;
+          if (buttonIndex == CONFIRM_RESTART_PROMPT_RESTART_NOW) {
+            Services.startup.quit(
+              Ci.nsIAppStartup.eAttemptQuit | Ci.nsIAppStartup.eRestart
+            );
+          }
+        });
+      }
+    });
+
     if (
       /^https?$/.test(uri.scheme) &&
       Services.prefs.getBoolPref("browser.chrome.site_icons")
diff --git a/browser/components/translations/content/fullPageTranslationsPanel.js b/browser/components/translations/content/fullPageTranslationsPanel.js
index f437dce2be..b7756bdc4c 100644
--- a/browser/components/translations/content/fullPageTranslationsPanel.js
+++ b/browser/components/translations/content/fullPageTranslationsPanel.js
@@ -1298,6 +1298,15 @@ var FullPageTranslationsPanel = new (class {
     );
   }
 
+  openDisableTranslations() {
+    const window =
+      gBrowser.selectedBrowser.browsingContext.top.embedderElement.ownerGlobal;
+    window.openTrustedLinkIn(
+      "about:preferences#general-translations-enable",
+      "tab"
+    );
+  }
+
   /**
    * Redirect the user to about:preferences
    */
