diff --git a/browser/components/urlbar/QuickActionsLoaderDefault.sys.mjs b/browser/components/urlbar/QuickActionsLoaderDefault.sys.mjs
index 691fcea3a1..6371e79a54 100644
--- a/browser/components/urlbar/QuickActionsLoaderDefault.sys.mjs
+++ b/browser/components/urlbar/QuickActionsLoaderDefault.sys.mjs
@@ -112,6 +112,8 @@ const DEFAULT_ACTIONS = {
     icon: "chrome://global/skin/icons/highlights.svg",
     label: "quickactions-manageai",
     onPick: openUrlFun("about:preferences#ai"),
+    isUnsupported: () =>
+      !Services.prefs.getBoolPref("browser.preferences.aiControls", false),
   },
   downloads: {
     l10nCommands: ["quickactions-cmd-downloads"],
@@ -337,6 +339,7 @@ const DEFAULT_ACTIONS = {
     l10nCommands: ["quickactions-cmd-labs"],
     icon: "chrome://global/skin/icons/experiments.svg",
     label: "quickactions-labs",
+    isUnsupported: () => !Services.policies.isAllowed("FirefoxLabs"),
     onPick: openUrlFun("about:preferences#experimental"),
   },
 };
