diff --git a/config/createprecomplete.py b/config/createprecomplete.py
index 07139f2..8e80807 100644
--- a/config/createprecomplete.py
+++ b/config/createprecomplete.py
@@ -27,7 +27,7 @@ def get_build_entries(root_path):
                 or rel_path_file.startswith("ChannelPrefs.framework/")
                 or "/UpdateSettings.framework/" in rel_path_file
                 or rel_path_file.startswith("UpdateSettings.framework/")
-                or "distribution/" in rel_path_file
+                or ("distribution/" in rel_path_file and rel_path_file != "distribution/policies.json")
             ):
                 rel_file_path_set.add(rel_path_file)
 
