diff --git a/browser/components/preferences/main.js b/browser/components/preferences/main.js
index 61af487f6a..947f0570ee 100644
--- a/browser/components/preferences/main.js
+++ b/browser/components/preferences/main.js
@@ -541,21 +541,19 @@ Preferences.addSetting(
 Preferences.addSetting({
   id: "linkPreviewEnabled",
   pref: "browser.ml.linkPreview.enabled",
-  deps: ["aiControlDefault", "aiControlLinkPreviews"],
-  visible: ({ aiControlDefault, aiControlLinkPreviews }) => {
-    return (
-      canShowAiFeature(aiControlLinkPreviews, aiControlDefault) &&
-      // @ts-ignore bug 1996860
-      LinkPreview.canShowPreferences
-    );
-  },
 });
 Preferences.addSetting({
   id: "linkPreviewKeyPoints",
   pref: "browser.ml.linkPreview.optin",
+  deps: ["aiControlDefault", "aiControlLinkPreviews"],
   // LinkPreview.canShowKeyPoints depends on the global genai pref.
-  // @ts-ignore bug 1996860
-  visible: () => LinkPreview.canShowKeyPoints,
+  visible: ({ aiControlDefault, aiControlLinkPreviews }) => {
+       return (
+         canShowAiFeature(aiControlLinkPreviews, aiControlDefault) &&
+         // @ts-ignore bug 1996860
+         LinkPreview.canShowKeyPoints
+       );
+  }
 });
 Preferences.addSetting({
   id: "linkPreviewLongPress",
