diff --git a/browser/components/migration/FirefoxProfileMigrator.sys.mjs b/browser/components/migration/FirefoxProfileMigrator.sys.mjs
index f932e4a011..2bd44a2081 100644
--- a/browser/components/migration/FirefoxProfileMigrator.sys.mjs
+++ b/browser/components/migration/FirefoxProfileMigrator.sys.mjs
@@ -13,6 +13,8 @@ import { MigrationUtils } from "resource:///modules/MigrationUtils.sys.mjs";
 
 import { MigratorBase } from "resource:///modules/MigratorBase.sys.mjs";
 
+import { AppConstants } from "resource://gre/modules/AppConstants.sys.mjs";
+
 const lazy = {};
 
 ChromeUtils.defineESModuleGetters(lazy, {
@@ -33,7 +35,7 @@ ChromeUtils.defineESModuleGetters(lazy, {
  */
 export class FirefoxProfileMigrator extends MigratorBase {
   static get key() {
-    return "firefox";
+    return AppConstants.MOZ_APP_NAME;
   }
 
   static get displayNameL10nID() {
diff --git a/browser/components/migration/FirefoxSelectableProfileMigrator.sys.mjs b/browser/components/migration/FirefoxSelectableProfileMigrator.sys.mjs
index ee90e0ce1a..6f7af20a96 100644
--- a/browser/components/migration/FirefoxSelectableProfileMigrator.sys.mjs
+++ b/browser/components/migration/FirefoxSelectableProfileMigrator.sys.mjs
@@ -12,6 +12,7 @@
 import { MigrationUtils } from "resource:///modules/MigrationUtils.sys.mjs";
 import { SelectableProfileService } from "resource:///modules/profiles/SelectableProfileService.sys.mjs";
 import { FirefoxProfileMigrator } from "resource:///modules/FirefoxProfileMigrator.sys.mjs";
+import { AppConstants } from "resource://gre/modules/AppConstants.sys.mjs";
 
 /**
  * Firefox SelectableProfile migrator. Currently, this class only does
@@ -24,7 +25,7 @@ import { FirefoxProfileMigrator } from "resource:///modules/FirefoxProfileMigrat
  */
 export class FirefoxSelectableProfileMigrator extends FirefoxProfileMigrator {
   static get key() {
-    return "firefox-selectable-profile";
+    return AppConstants.MOZ_APP_NAME + "-selectable-profile";
   }
 
   /**
diff --git a/browser/components/migration/MigrationUtils.sys.mjs b/browser/components/migration/MigrationUtils.sys.mjs
index a05920b847..73da14f1a6 100644
--- a/browser/components/migration/MigrationUtils.sys.mjs
+++ b/browser/components/migration/MigrationUtils.sys.mjs
@@ -44,8 +44,8 @@ function getL10n() {
 }
 
 const FIREFOX_REFRESH_MIGRATOR_KEYS = new Set([
-  "firefox",
-  "firefox-selectable-profile",
+  AppConstants.MOZ_APP_NAME,
+  AppConstants.MOZ_APP_NAME + "-selectable-profile",
 ]);
 
 const MIGRATOR_MODULES = Object.freeze({
@@ -509,7 +509,7 @@ class MigrationUtils {
       "Internet Explorer": "ie",
       "Microsoft Edge": "edge",
       Safari: "safari",
-      Firefox: "firefox",
+      [AppConstants.MOZ_APP_BASENAME]: AppConstants.MOZ_APP_NAME,
       Nightly: "firefox",
       Opera: "opera",
       Vivaldi: "vivaldi",
@@ -530,8 +530,8 @@ class MigrationUtils {
         .getApplicationDescription("http");
       key = APP_DESC_TO_KEY[browserDesc] || "";
       // Handle devedition, as well as "FirefoxNightly" on OS X.
-      if (!key && browserDesc.startsWith("Firefox")) {
-        key = "firefox";
+      if (!key && browserDesc.startsWith(AppConstants.MOZ_APP_BASENAME)) {
+        key = AppConstants.MOZ_APP_NAME;
       }
     } catch (ex) {
       console.error("Could not detect default browser: ", ex);
diff --git a/browser/locales/en-US/browser/migrationWizard.ftl b/browser/locales/en-US/browser/migrationWizard.ftl
index 8341f95a6f..2731e610b5 100644
--- a/browser/locales/en-US/browser/migrationWizard.ftl
+++ b/browser/locales/en-US/browser/migrationWizard.ftl
@@ -36,7 +36,7 @@ migration-wizard-migrator-display-name-chromium-360se = 360 Secure Browser
 migration-wizard-migrator-display-name-chromium-edge = Microsoft Edge
 migration-wizard-migrator-display-name-chromium-edge-beta = Microsoft Edge Beta
 migration-wizard-migrator-display-name-edge-legacy = Microsoft Edge Legacy
-migration-wizard-migrator-display-name-firefox = Firefox
+migration-wizard-migrator-display-name-firefox = { -brand-product-name }
 migration-wizard-migrator-display-name-file-password-csv = Passwords from CSV file
 migration-wizard-migrator-display-name-file-bookmarks = Bookmarks from HTML file
 migration-wizard-migrator-display-name-ie = Microsoft Internet Explorer
