diff --git a/toolkit/components/extensions/parent/ext-runtime.js b/toolkit/components/extensions/parent/ext-runtime.js
index abc30a2..bda1643 100644
--- a/toolkit/components/extensions/parent/ext-runtime.js
+++ b/toolkit/components/extensions/parent/ext-runtime.js
@@ -334,7 +334,16 @@ this.runtime = class extends ExtensionAPIPersistent {
 
         getBrowserInfo: function () {
           const { name, vendor, version, appBuildID } = Services.appinfo;
-          const info = { name, vendor, version, buildID: appBuildID };
+          const useFirefoxDefaults = Services.prefs.getBoolPref(
+            "librewolf.getBrowserInfo.setToFirefoxDefaults",
+            true
+          );
+
+          const version_display = AppConstants.MOZ_APP_VERSION_DISPLAY;
+          const info = useFirefoxDefaults
+            ? { name: "Firefox", vendor: "Mozilla", version, buildID: appBuildID }
+            : { name, vendor, version: version_display, buildID: appBuildID };
+
           return Promise.resolve(info);
         },
 
diff --git a/toolkit/xre/nsAppRunner.cpp b/toolkit/xre/nsAppRunner.cpp
index 1cf5753..7c88751 100644
--- a/toolkit/xre/nsAppRunner.cpp
+++ b/toolkit/xre/nsAppRunner.cpp
@@ -2256,9 +2256,6 @@ static void DumpHelp() {
 }
 
 static inline void DumpVersion() {
-  if (gAppData->vendor && *gAppData->vendor) {
-    printf("%s ", (const char*)gAppData->vendor);
-  }
   printf("%s ", (const char*)gAppData->name);
 
   // Use the displayed version
@@ -2272,9 +2269,6 @@ static inline void DumpVersion() {
 }
 
 static inline void DumpFullVersion() {
-  if (gAppData->vendor && *gAppData->vendor) {
-    printf("%s ", (const char*)gAppData->vendor);
-  }
   printf("%s ", (const char*)gAppData->name);
 
   // Use the displayed version
