diff --git a/browser/components/preferences/jar.mn b/browser/components/preferences/jar.mn
index 118709048e..51aeb42a23 100644
--- a/browser/components/preferences/jar.mn
+++ b/browser/components/preferences/jar.mn
@@ -11,6 +11,7 @@ browser.jar:
    content/browser/preferences/home.js
    content/browser/preferences/search.js
    content/browser/preferences/privacy.js
+   content/browser/preferences/librewolf.js
    content/browser/preferences/containers.js
    content/browser/preferences/translations.js
    content/browser/preferences/sync.js
diff --git a/browser/components/preferences/preferences.js b/browser/components/preferences/preferences.js
index e4746e3408..3755be8a78 100644
--- a/browser/components/preferences/preferences.js
+++ b/browser/components/preferences/preferences.js
@@ -9,6 +9,7 @@
 /* import-globals-from containers.js */
 /* import-globals-from translations.js */
 /* import-globals-from privacy.js */
+/* import-globals-from librewolf.js */
 /* import-globals-from sync.js */
 /* import-globals-from experimental.js */
 /* import-globals-from moreFromMozilla.js */
@@ -196,6 +197,7 @@ function init_all() {
   register_module("paneHome", gHomePane);
   register_module("paneSearch", gSearchPane);
   register_module("panePrivacy", gPrivacyPane);
+  register_module("paneLibrewolf", gLibrewolfPane);
   register_module("paneContainers", gContainersPane);
 
   if (Services.prefs.getBoolPref("browser.translations.newSettingsUI.enable")) {
diff --git a/browser/components/preferences/preferences.xhtml b/browser/components/preferences/preferences.xhtml
index 43be2b6ac8..78e9bfe676 100644
--- a/browser/components/preferences/preferences.xhtml
+++ b/browser/components/preferences/preferences.xhtml
@@ -42,6 +42,7 @@
     href="chrome://browser/skin/preferences/containers.css"
   />
   <link rel="stylesheet" href="chrome://browser/skin/preferences/privacy.css" />
+  <link rel="stylesheet" href="chrome://browser/skin/preferences/librewolf.css" />
   <link rel="stylesheet" href="chrome://browser/content/preferences/widgets/setting-group.css" />
 
   <link rel="localization" href="branding/brand.ftl"/>
@@ -147,6 +148,17 @@
           <label class="category-name" flex="1" data-l10n-id="pane-privacy-title"></label>
         </richlistitem>
 
+        <richlistitem id="category-librewolf"
+                      class="category"
+                      value="paneLibrewolf"
+                      helpTopic="prefs-librewolf"
+                      data-l10n-id="category-librewolf"
+                      data-l10n-attrs="tooltiptext"
+                      align="center">
+          <image class="category-icon"/>
+          <label class="category-name" flex="1" data-l10n-id="pane-librewolf-title"></label>
+        </richlistitem>
+
         <richlistitem id="category-sync"
                       class="category"
                       hidden="true"
@@ -241,6 +253,7 @@
 #include home.inc.xhtml
 #include search.inc.xhtml
 #include privacy.inc.xhtml
+#include librewolf.inc.xhtml
 #include containers.inc.xhtml
 #include translations.inc.xhtml
 #include sync.inc.xhtml
diff --git a/browser/themes/shared/jar.inc.mn b/browser/themes/shared/jar.inc.mn
index c2bebf7e96..62ab632cec 100644
--- a/browser/themes/shared/jar.inc.mn
+++ b/browser/themes/shared/jar.inc.mn
@@ -114,6 +114,7 @@
   skin/classic/browser/preferences/android-menu.svg            (../shared/preferences/android-menu.svg)
   skin/classic/browser/preferences/category-general.svg        (../shared/preferences/category-general.svg)
   skin/classic/browser/preferences/category-privacy-security.svg    (../shared/preferences/category-privacy-security.svg)
+  skin/classic/browser/preferences/category-librewolf.svg    (../shared/preferences/category-librewolf.svg)
   skin/classic/browser/preferences/category-search.svg         (../shared/preferences/category-search.svg)
   skin/classic/browser/preferences/category-sync.svg           (../shared/preferences/category-sync.svg)
   skin/classic/browser/preferences/containers.css              (../shared/preferences/containers.css)
@@ -131,6 +132,7 @@
   skin/classic/browser/preferences/vpn-logo.svg                (../shared/preferences/vpn-logo.svg)
   skin/classic/browser/preferences/search.css                  (../shared/preferences/search.css)
   skin/classic/browser/preferences/siteDataSettings.css        (../shared/preferences/siteDataSettings.css)
+  skin/classic/browser/preferences/librewolf.css               (../shared/preferences/librewolf.css)
   skin/classic/browser/translations/panel.css                  (../shared/translations/panel.css)
   skin/classic/browser/fxa/fxa-spinner.svg                     (../shared/fxa/fxa-spinner.svg)
   skin/classic/browser/fxa/sync-illustration.svg               (../shared/fxa/sync-illustration.svg)
diff --git a/browser/themes/shared/preferences/preferences.css b/browser/themes/shared/preferences/preferences.css
index 641c936533..7ff76e622f 100644
--- a/browser/themes/shared/preferences/preferences.css
+++ b/browser/themes/shared/preferences/preferences.css
@@ -223,6 +223,10 @@ radio {
   list-style-image: url("chrome://browser/skin/preferences/category-privacy-security.svg");
 }
 
+#category-librewolf > .category-icon {
+  list-style-image: url("chrome://browser/skin/preferences/category-librewolf.svg");
+}
+
 #category-sync > .category-icon {
   list-style-image: url("chrome://browser/skin/preferences/category-sync.svg");
 }
