diff --git a/browser/components/preferences/main.inc.xhtml b/browser/components/preferences/main.inc.xhtml
--- a/browser/components/preferences/main.inc.xhtml
+++ b/browser/components/preferences/main.inc.xhtml
@@ -21,9 +21,6 @@
   <html:h1 data-l10n-id="pane-general-title"/>
 </hbox>
 
-<!-- Default browser -->
-<html:setting-group groupid="defaultBrowser" data-category="paneGeneral" hidden="true" />
-
 <!-- Startup -->
 <html:setting-group id="startupGroup" groupid="startup" data-category="paneGeneral" hidden="true"/>
 
diff --git a/browser/components/preferences/main.js b/browser/components/preferences/main.js
--- a/browser/components/preferences/main.js
+++ b/browser/components/preferences/main.js
@@ -2643,7 +2643,6 @@
       },
     ],
   },
-  defaultBrowser: createDefaultBrowserConfig(),
   startup: {
     l10nId: "startup-group",
     headingLevel: 2,
@@ -2682,10 +2681,6 @@
         id: "windowsLaunchOnLoginDisabledProfileBox",
         control: "moz-message-bar",
         l10nId: "startup-windows-launch-on-login-profile-disabled",
-      },
-      {
-        id: "alwaysCheckDefault",
-        l10nId: "always-check-default",
       },
     ],
   },
@@ -5354,7 +5349,6 @@
     initSettingGroup("translations");
     initSettingGroup("spellCheck");
     initSettingGroup("performance");
-    initSettingGroup("defaultBrowser");
     initSettingGroup("startup");
     initSettingGroup("importBrowserData");
     initSettingGroup("networkProxy");