diff --git a/browser/components/preferences/main.inc.xhtml b/browser/components/preferences/main.inc.xhtml
index 3b07de78a6..f27cf7ff78 100644
--- a/browser/components/preferences/main.inc.xhtml
+++ b/browser/components/preferences/main.inc.xhtml
@@ -27,10 +27,13 @@
   <label><html:h2 data-l10n-id="startup-header"/></label>
 
   <vbox id="startupPageBox">
-    <html:setting-group groupid="startup"/>
-    <html:a is="moz-support-link" support-page="session-restore" data-l10n-id="session-restore-learn-more" />
+  <html:setting-group groupid="startup"/>
+  <html:a is="moz-support-link"
+          support-page="session-restore"
+          data-l10n-id="session-restore-learn-more"/>
   </vbox>
 
+
 </groupbox>
 
 <!-- Data migration -->
