diff --git a/browser/components/preferences/main.js b/browser/components/preferences/main.js
index 46283d3eac..cd40b7bd85 100644
--- a/browser/components/preferences/main.js
+++ b/browser/components/preferences/main.js
@@ -525,9 +529,69 @@ Preferences.addSetting({
   id: "web-appearance-override-warning",
   setup: emitChange => {
     FORCED_COLORS_QUERY.addEventListener("change", emitChange);
-    return () => FORCED_COLORS_QUERY.removeEventListener("change", emitChange);
+    const prefObserver = {
+      observe(subject, topic, data) {
+        emitChange();
+      },
+    };
+    Services.prefs.addObserver("privacy.resistFingerprinting", prefObserver);
+    return () => {
+      FORCED_COLORS_QUERY.removeEventListener("change", emitChange);
+      Services.prefs.removeObserver("privacy.resistFingerprinting", prefObserver);
+    };
   },
   visible: () => {
+    const warning = document.getElementById("web-appearance-override-warning");
+    if (warning) {
+      warning.hidden = !FORCED_COLORS_QUERY.matches;
+    }
+
+    document.getElementById("web-appearance-rfp-warning")?.remove();
+
+    const chooser = document.getElementById("web-appearance-chooser");
+    const webAppearanceSettings = document.getElementById("webAppearanceSettings");
+
+    if (Services.prefs.getBoolPref("privacy.resistFingerprinting")) {
+      if (chooser) {
+        chooser.style.opacity = 0.3;
+        chooser.style.pointerEvents = "none";
+      }
+
+      
+      const infoBox = document.createElement("div");
+      infoBox.id = "web-appearance-rfp-warning";
+      infoBox.classList.add("info-box-container");
+      infoBox.style.display = "flex";
+      infoBox.style.alignItems = "center";
+      infoBox.style.marginTop = "10px";
+      infoBox.style.marginBottom = "5px";
+
+      const text = document.createElement("div");
+      text.innerText =
+        "This feature is disabled because ResistFingerprinting is enabled. This means LibreWolf will force web content to display in a light theme.";
+      infoBox.appendChild(text);
+
+      const learnMore = document.createElement("a");
+      learnMore.classList.add("text-link");
+      learnMore.style.marginLeft = "5px";
+      learnMore.style.flexShrink = 0;
+      learnMore.setAttribute("is", "learn-more");
+      learnMore.href = "https://librewolf.net/docs/faq/#why-is-librewolf-forcing-light-theme";
+      learnMore.innerText = "Learn more";
+      infoBox.appendChild(learnMore);
+
+      
+      if (webAppearanceSettings) {
+        const beforeNode = webAppearanceSettings.children[2] || null;
+        webAppearanceSettings.insertBefore(infoBox, beforeNode);
+      }
+    } else {
+      if (chooser) {
+        chooser.style.opacity = 1;
+        chooser.style.pointerEvents = "all";
+      }
+    }
+    
     return FORCED_COLORS_QUERY.matches;
   },
 });
