diff --git a/browser/components/preferences/main.js b/browser/components/preferences/main.js
index 0879c5e10d..e09fab02f6 100644
--- a/browser/components/preferences/main.js
+++ b/browser/components/preferences/main.js
@@ -759,6 +759,27 @@ Preferences.addSetting({
   },
 });
 
+Preferences.addSetting({
+  id: "librewolfRFPWarning",
+  deps: ["librewolfRFP"],
+  visible: ({librewolfRFP}) => {
+    const interval = setInterval(() => {
+      const chooser = document.getElementById("web-appearance-chooser");
+      if (chooser) {
+        clearInterval(interval);
+        if (librewolfRFP.value) {
+          chooser.style.opacity = "0.3";
+          chooser.style.pointerEvents = "none";
+        } else {
+          chooser.style.opacity = "1";
+          chooser.style.pointerEvents = "all";
+        }
+      }
+    }, 100);
+    return librewolfRFP.value;
+  },
+});
+
 Preferences.addSetting(
   /** @type {{ themeNames: string[] } & SettingConfig}} */ ({
     id: "web-appearance-chooser",
@@ -2719,6 +2727,12 @@ SettingGroupManager.registerGroups({
         l10nId: "preferences-web-appearance-override-warning3",
         control: "moz-message-bar",
       },
+      {
+        id: "librewolfRFPWarning",
+        l10nId: "librewolf-rfp-warning",
+        supportPage: "rfp-theme",
+        control: "moz-message-bar",
+      },
       {
         id: "web-appearance-chooser",
         control: "moz-visual-picker",
