diff --git a/browser/components/preferences/config/appearance.mjs b/browser/components/preferences/config/appearance.mjs
index 6a2a200..14fea29 100644
--- a/browser/components/preferences/config/appearance.mjs
+++ b/browser/components/preferences/config/appearance.mjs
@@ -94,6 +94,27 @@ Preferences.addSetting({

 Preferences.addSetting({ id: "relatedSettingsBoxGroup" });
 
+Preferences.addSetting({
+  id: "librewolfRFPWarning",
+  deps: ["librewolfRFP"],
+  visible: ({librewolfRFP}) => {
+    const interval = setInterval(() => {
+      const chooser = document.getElementById("web-appearance-chooser");
+      if (chooser) {
+        clearInterval(interval);
+        if (librewolfRFP.value) {
+          chooser.style.opacity = "0.3";
+          chooser.style.pointerEvents = "none";
+        } else {
+          chooser.style.opacity = "1";
+          chooser.style.pointerEvents = "all";
+        }
+      }
+    }, 100);
+    return librewolfRFP.value;
+  },
+});
+
 SettingGroupManager.registerGroups({
   appearance: {
     l10nId: "appearance-group2",
@@ -108,6 +129,12 @@ SettingGroupManager.registerGroups({
           role: "status",
         },
       },
+      {
+        id: "librewolfRFPWarning",
+        l10nId: "librewolf-rfp-warning",
+        supportPage: "rfp-theme",
+        control: "moz-message-bar",
+      },
       {
         id: "web-appearance-chooser",
         control: "moz-visual-picker",
